bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/fs/watch/JSDependencyTracker.java

```diff
@@ -31,32 +32,23 @@
  * Tracks JS module dependencies
  *
  * @author Jonathan Gilbert - Initial contribution
+ * @author Jan N. Klug - Refactored to new WatchService
  */
 @Component(service = JSDependencyTracker.class)
 @NonNullByDefault
 public class JSDependencyTracker extends AbstractScriptDependencyTracker {
 
     private final Logger logger = LoggerFactory.getLogger(JSDependencyTracker.class);
 
-    public static final String LIB_PATH = String.join(File.separator, OpenHAB.getConfigFolder(), "automation", "js",
-            "node_modules");
-
-    public JSDependencyTracker() {
-        super(LIB_PATH);
-    }
+    public static final String LIB_PATH = String.join(File.separator, "automation", "js", "node_modules");
 
     @Activate
-    public void activate() {
-        File directory = new File(LIB_PATH);
-        if (!directory.exists()) {
-            if (!directory.mkdirs()) {
-                logger.warn("Failed to create watched directory: {}", LIB_PATH);
-            }
-        } else if (directory.isFile()) {
-            logger.warn("Trying to watch directory {}, however it is a file", LIB_PATH);
-        }
+    public JSDependencyTracker(@Reference(target = WatchService.CONFIG_WATCHER_FILTER) WatchService watchService) {
+        super(watchService, LIB_PATH);
 
-        super.activate();
+        if (Files.isRegularFile(this.libraryPath)) {
```

Comment:
```suggestion
        if (Files.isRegularFile(this.LIB_PATH)) {
```

Reply:
Debatable if it is more clear if we use `libraryPath` which is what is used by the `WatchService` itself or `Path.of(LIB_PATH)`. Technically it is the same. The constructor calls `super(watchService, LIB_PATH)` and there `libraryPath` is assigned to `Path.of(LIB_PATH)`.